pkg/util/duration/duration.go

```diff
@@ -77,6 +86,80 @@ func (d Duration) Compare(x Duration) int {
 	return 0
 }
 
+// FromInt64 converts an int64 number of seconds to a
+// duration. Inverse conversion of AsInt64.
+func FromInt64(x int64) Duration {
+	days := x / (nanosInDay / nanosInSecond)
+	seconds := x % (nanosInDay / nanosInSecond)
+	d := Duration{Days: days, Nanos: seconds * nanosInSecond}
+	return d.normalize()
+}
+
+// FromFloat64 converts a float64 number of microseconds to a
```

Comment:
this comment says microseconds. is that correct?

Reply:
Woops, thanks. Corrected.